test/integration/build/test_build.py

```diff
@@ -81,6 +81,113 @@ def test_app_without_rockspec(cartridge_cmd, project_without_dependencies):
     assert 'Application directory should contain rockspec' in output
 
 
+def test_app_with_rockspec_bad_name(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    bad_name_rockspec = "bad_rockspec-scm-1.rockspec"
+    rocks_make_output = "Rockspec %s doesn't exist" % (bad_name_rockspec)
+
+    # with --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        bad_name_rockspec,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 1, 'Building project should fail'
+    assert rocks_make_output in output
+
+
+def test_app_with_rockspec_from_other_dir(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    dir_name = 'some_dir'
+    dir_path = os.path.join(project.path, dir_name)
+    os.mkdir(dir_path)
+
+    version = 'scm-2'
+    rockspec_name = '%s-%s.rockspec' % (project.name, version)
+    rockspec_path = os.path.join(dir_path, rockspec_name)
+    who_am_i = 'I am %s' % rockspec_name
+
+    with open(rockspec_path, 'w') as f:
```

Comment:
What about placing this multiple reused code in `project` module? To simply say `set_whoami_build_specs(project, ['scm-1', 'scm-2'])`.